### PR TITLE
Support exporting billing report to CSV and Excel

### PR DESCRIPTION
**Pull request by vboctor**
- Added billing_api.php
- Implemented 'Export to Excel' using billing api
- Implemented 'Export to CSV' using billing api

Will open a code cleanup bug to move more code to use billing_api.

Fixes #7932

### CODE REVIEW

**Comment by dregad** [comment]
2 high-level comments, without looking at the code...

- I don't think we should be spending effort on a deprecated feature [[1]](http://thread.gmane.org/gmane.comp.bug-tracking.mantis.devel/3348/focus=3355),[[2]](https://www.mantisbt.org/blog/?p=416)
- Now that we're in release candidate mode, it may be best to postpone adding a new API to introduce new features until after 1.3.0 goes live.

That being said, the decision to deprecate time tracking was not mine, and I don't feel strongly about maintaining it even though I do agree that its design is dated and functionality limited. There are a few plugins out there which do a much better job at it.

**Comment by vboctor** [comment]
I think the export functionality is useful specially if we are going to deprecate the feature.  I haven't done a comparison with existing plugins, so not sure about how they compare.

As for locking down 1.3.0, I don't see this change as disruptive, as you can see, other than adding the link to the billing page, the changes are mostly new APIs to provide the export functionality.  I think we will keep allowing non-disruptive changes to master until we fork master-1.3.x.

File: core/billing_api.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+	access_ensure_project_level( config_get( 'time_tracking_reporting_threshold' ), $p_project_id, $p_user_id );
+}
+
+/**
+ * Returns an array of bugnote stats
+ * @param integer $p_project_id A project identifier.
+ * @param string  $p_from       Starting date (yyyy-mm-dd) inclusive, if blank, then ignored.
+ * @param string  $p_to         Ending date (yyyy-mm-dd) inclusive, if blank, then ignored.
+ * @param integer $p_cost       Cost.
```
Comment (atrol) [inline core/billing_api.php]: $p_cost_per_hour

Comment (vboctor) [inline core/billing_api.php]: Fixed.

**Comment by vboctor** [comment]
@atrol I've addressed your comments.
@dregad let me know your feedback.

**Comment by dregad** [comment]
In terms of implementation, considering that CSV vs Excel export are nearly identical in terms of structure, I would personally have gone for an object-based approach (e.g. having a *BillingExport* class and 2 child classes *BillingExportExcel* and *BillingExportCsv*) to reduce code duplication.

That being said, I don't mind if you merge this.

**Pull request merged**